torch/_inductor/utils.py

```diff
@@ -987,6 +988,41 @@ def use_cutlass_template(layout, m, n, k):
     return res
 
 
+def use_ck_template(layout, m, n, k):
+    # config knobs check 1
+    if not use_max_autotune():
+        return False
+    # config knobs check 2
+    if not _use_autotune_backend("CK"):
+        return False
+    # platform check
+    if not torch.version.hip:
+        return False
+    # tensors must be on GPU
+    if not layout.device.type == "cuda":
+        return False
+    # hardware check, enable for CDNA3 only for now
+    device_capability = torch.cuda.get_device_capability(layout.device)
+    log.debug("Device capability %s", device_capability)
+    # https://llvm.org/docs/AMDGPUUsage.html#amd-gcn-gfx940-gfx942-cdna3
+    if device_capability != (9, 4):
+        return False
+    # supported input dtypes
+    if layout.dtype not in [torch.float16, torch.bfloat16]:
+        return False
+    # TBD investigate if we need to disable backend based on number of available CUs
+    # if not is_big_gpu(layout.device.index or 0):
+    #     return False
+    from .virtualized import V
+
+    # check if shape is static and gemm size is not 0
+    gemm_size = V.graph.sizevars.size_hint(m * n * k, fallback=-1)
```

Comment:
Hmm not sure if I understand your point. My intent here is to evaluate if the shape is static, and, iiuc, `size_hint` returns 0 or (M*N*K) if the shape is static, or returns `-1` if it's not static. Then, if the result is -1 (not static) or 0 (one of M, N, K is zero) we cannot proceed and return False. Otherwise the check is passed. 

Also the CUTLASS backend has the same check in https://github.com/pytorch/pytorch/blob/main/torch/_inductor/utils.py#L965-L967